projects/core/components/data-list/data-list.component.ts

```diff
@@ -63,7 +69,7 @@ export class TuiDataListComponent<T> implements TuiDataListAccessor<T> {
 
     @Input()
     @HostBinding('attr.data-list-size')
-    public size = this.controller?.size || 'm';
+    public size: TuiSizeL | TuiSizeS | TuiSizeXS = this.controller?.size || 'm';
```

Comment:
TuiSizeS is unnecessary, it is included in TuiSizeXS

Reply:
fixed